crates/compiler/can/src/constraint.rs

```diff
@@ -773,6 +784,10 @@ pub enum Constraint {
     /// Attempt to resolve a specialization.
     Resolve(OpportunisticResolve),
     CheckCycle(Index<Cycle>, IllegalCycleMark),
+
+    // This is terrible and could be a huge cost to copy.
+    // Not sure a better way to get the bytes here so we can check if they are valid utf8 or decode properly.
+    IngestedFile(TypeOrVar, Box<Path>, Vec<u8>),
```

Comment:
I think that this containing a `Vec<u8>` is terrible. That said, we need the bytes in solve. That is where we check that is where we will check for valid utf8 if it is a string. Also, in a future with decode, I guess that may be where we check that the bytes decode successful. @ayazhafiz Any ideas on how to avoid copying all the bytes here? I guess I could try adding a type variable to Constraint and piping that around to get it as a slice?

Reply:
I guess another option would be to delay those errors until when we manifest the final type when converting to Mono. That said, I don't think converting to mono can currently error, so we would need to build up the error somehow.

Reply:
it would be better to do this in solve.

Maybe an `Rc<Vec<u8>>` for now?

Reply:
Yeah, that works to avoid the copy. Switched.